dronegen/push.go

```diff
@@ -1,6 +1,175 @@
 package main
 
+import "fmt"
+
 func pushPipelines() []pipeline {
-	// TODO: migrate
-	return nil
+	return []pipeline{
+		pushPipeline(buildType{os: "linux", arch: "amd64"}),
+		pushPipeline(buildType{os: "linux", arch: "i386"}),
+		pushPipeline(buildType{os: "linux", arch: "amd64", fips: true}),
+		pushPipeline(buildType{os: "windows", arch: "amd64"}),
+		pushPipeline(buildType{os: "linux", arch: "arm"}),
+		pushPipeline(buildType{os: "linux", arch: "arm", fips: true}),
+		pushPipeline(buildType{os: "linux", arch: "arm64"}),
+		pushPipeline(buildType{os: "linux", arch: "arm64", fips: true}),
+	}
+}
+
+// translatePipelineArch converts a nominal arch to a Go architecture representation
+// e.g. i386 -> 386, x86_64 -> amd64
+// This is required because the existing pipeline uses i386 whereas `GOARCH` is expected to be 386
+func translatePipelineArch(arch string) string {
+	if arch == "i386" {
+		return "386"
+	}
+	return arch
+}
+
+func pushBuildCommands(params buildType) []string {
+	commands := []string{
+		`apk add --no-cache make`,
+		`chown -R $UID:$GID /go`,
+	}
+	makefileTarget := "release"
+	baseDockerImage := "quay.io/gravitational/teleport-buildbox:$RUNTIME"
+	var extraDockerImage string
+	var makeCommand string
+
+	if params.fips {
+		makefileTarget = "release-fips"
+		baseDockerImage = "quay.io/gravitational/teleport-buildbox-fips:$RUNTIME"
+		if params.arch == "arm" {
+			makefileTarget = "release-arm-fips"
+			extraDockerImage = "quay.io/gravitational/teleport-buildbox-arm-fips:$RUNTIME"
+		} else if params.arch == "arm64" {
+			makefileTarget = "release-arm64-fips"
+			extraDockerImage = "quay.io/gravitational/teleport-buildbox-arm-fips:$RUNTIME"
+		}
+		makeCommand = fmt.Sprintf(`make -C build.assets %s VERSION=$VERSION OS=$OS ARCH=$ARCH FIPS=$FIPS`, makefileTarget)
+	} else {
+		if params.arch == "arm" {
+			makefileTarget = "release-arm"
+			extraDockerImage = "quay.io/gravitational/teleport-buildbox-arm:$RUNTIME"
+		} else if params.arch == "arm64" {
+			makefileTarget = "release-arm64"
+			extraDockerImage = "quay.io/gravitational/teleport-buildbox-arm:$RUNTIME"
+		}
+		makeCommand = fmt.Sprintf(`make -C build.assets %s OS=$OS ARCH=$ARCH`, makefileTarget)
+	}
```

Comment:
I don't like this section much, I originally had a `switch (params.arch)` block but the FIPS part complicates it somewhat.

Reply:
How about we add alias make rules to make target naming consistent?
Add something like:
```
release-amd64: release
release-386: release
release-amd64-fips: release-fips
release-386-fips: release-fips
```
(maybe also force the `ARCH` value)

Then you could do:
```
makefileTarget := fmt.Sprintf("release-%s", params.arch)
if params.fips {
  makefileTarget += "-fips"
}
```

And something similar for docker images.

Reply:
This makes sense.